nixos/modules/services/web-apps/alps.nix

```diff
@@ -0,0 +1,99 @@
+{ lib, pkgs, config, ... }:
+
+with lib;
+
+let
+  cfg = config.services.alps;
+in {
+  options.services.alps = {
+    enable = mkEnableOption "alps";
+
+    port = mkOption {
+      type = types.port;
+      default = 1323;
+      description = ''
+        TCP port the process should listen to.
+      '';
+    };
+
+    bindIP = mkOption {
+      default = "[::]";
+      type = types.str;
+      description = ''
+        IP the service should listen to.
+      '';
+    };
+
+    theme = mkOption {
+      type = types.enum [ "alps" "sourcehut" ];
+      default = "sourcehut";
+      description = ''
+        The frontend's theme.
+      '';
+    };
+
+    imaps = {
```

Comment:
This whole part together with smtps below can be simplified into a single option `upstreams`, a list of string, to align with the upstream flags.

Reply:
> a list of string, to align with the upstream flags.

be careful with that, we need either a single argument or two, not more, so we'd need some validation here.